### PR TITLE
Parse data from url

### PR DESCRIPTION
**Pull request by YuriyHrechka**
Added new method that is parse valid data.

### CODE REVIEW

File: services.py [hunk 1]
```diff
@@ -23,5 +24,31 @@ def base_url_list(self) -> list:
             return urls
         raise ValueError('url is invalid')
 
+    def parse_data(self, url: str) -> dict:
+        response = requests.get(url=url, headers=self.headers)
+        data_dict = {}
+        valid_data = ['№ у системі:', 'Скорочена:', 'Поштова адреса:', 'Телефони:', 'E-mail:', 'Директор:', 'Спроможність закладу освіти (учнів):', 'Кількість учнів:']
```
Comment (MaksStud) [inline services.py:30]: This list and validation system must be another file and method

File: services.py [hunk 2]
```diff
@@ -23,5 +24,31 @@ def base_url_list(self) -> list:
...
+                th = row.find('th')
+                td = row.find('td')
+
+                if th and td:
+                    th_text = th.text.strip()
+                    td_text = td.text.strip()
+                    if th_text == 'E-mail:':
+                        encoded_email = soup.find('a', class_='static')['onclick'].split("unescape('")[1].split("')")[0]
+                        td_text = BeautifulSoup(urllib.parse.unquote(encoded_email), 'html.parser').find('a').text
+                    if th_text in valid_data:
```
Comment (MaksStud) [inline services.py:47]: Write another method from validation data, rename valid_data to validation_parameter_list

File: services.py [hunk 3]
```diff
@@ -23,5 +24,31 @@ def base_url_list(self) -> list:
...
+                        data_dict[th_text] = td_text
```
Comment (MaksStud) [inline services.py:48]: Use method from add

File: services.py [hunk 4]
```diff
@@ -23,5 +24,31 @@ def base_url_list(self) -> list:
...
+        if response.status_code == 200:
+            soup = BeautifulSoup(response.text, 'html.parser')
+            table = soup.find('table', class_="zebra-stripe")
+            rows = table.find_all('tr')
+
+            for row in rows:
+                th = row.find('th')
+                td = row.find('td')
+
+                if th and td:
```
Comment (MaksStud) [inline services.py:41]: Rename variables, this name is incorrect

File: main.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
 from services import SchoolParser
 
 a = SchoolParser('https://te.isuo.org/koatuu/schools-list/id/6110100000')
-print(a.base_url_list())
+print(a.parse_data(a.base_url_list()[0]))
```
Comment (MaksStud) [inline main.py:4]: Write a method from the parses full school from base_url_list

**Review by MaksStud — CHANGES_REQUESTED** [review]

File: services.py [hunk 6]
```diff
@@ -2,6 +2,8 @@
 from bs4 import BeautifulSoup
 import urllib.parse
 
+from utils import valid_data
```
Comment (YuriyHrechka) [inline services.py:5]: If we add new imports, we need to change requirements

**Pull request merged**